pkg/webhooks/mutator/virt-launcher-mutator.go

```diff
@@ -0,0 +1,234 @@
+		m.logInfo("not a pod creation - ignoring")
+		return admission.Allowed(ignoreOperationMessage)
+	}
+
+	launcherPod := &k8sv1.Pod{}
+	err := m.decoder.Decode(req, launcherPod)
+	if err != nil {
+		m.logErr(err, "cannot decode virt-launcher pod")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	originalPod := launcherPod.DeepCopy()
+
+	hco, err := getHcoObject(ctx, m.cli, m.hcoNamespace)
+	if err != nil {
+		m.logErr(err, "cannot get the HyperConverged object")
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	enforceCpuLimits, enforceMemoryLimits, err := m.getResourcesToEnforce(ctx, launcherPod.Namespace, hco)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if !enforceCpuLimits && !enforceMemoryLimits {
+		return admission.Allowed(ignoreOperationMessage)
+	}
+
+	if err := m.handleVirtLauncherCreation(launcherPod, hco, enforceCpuLimits, enforceMemoryLimits); err != nil {
+		m.logErr(err, "failed handling launcher pod %s", launcherPod.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	allowResponse := m.getAllowedResponseWithPatches(launcherPod, originalPod)
+	m.logInfo("mutation completed successfully for pod %s", launcherPod.Name)
+	return allowResponse
+}
+
+func (m *VirtLauncherMutator) handleVirtLauncherCreation(launcherPod *k8sv1.Pod, hco *v1beta1.HyperConverged, enforceCpuLimits, enforceMemoryLimits bool) error {
+	var cpuRatioStr, memRatioStr string
+	if enforceCpuLimits {
+		cpuRatioStr = hco.Annotations[cpuLimitToRequestRatioAnnotation]
+	}
+	if enforceMemoryLimits {
+		memRatioStr = hco.Annotations[memoryLimitToRequestRatioAnnotation]
+	}
+
+	if enforceCpuLimits {
+		err := m.setResourceRatio(launcherPod, cpuRatioStr, cpuLimitToRequestRatioAnnotation, k8sv1.ResourceCPU)
+		if err != nil {
+			return err
+		}
+	}
+	if enforceMemoryLimits {
+		err := m.setResourceRatio(launcherPod, memRatioStr, memoryLimitToRequestRatioAnnotation, k8sv1.ResourceMemory)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *VirtLauncherMutator) setResourceRatio(launcherPod *k8sv1.Pod, ratioStr, annotationKey string, resourceName k8sv1.ResourceName) error {
+	ratio, err := strconv.ParseFloat(ratioStr, 64)
+	if err != nil {
+		return fmt.Errorf("%s can't parse ratio %s to float: %w. The ratio is the value of annotation key %s", launcherMutatorStr, ratioStr, err, annotationKey)
+	}
+
+	if ratio <= 0 {
+		return fmt.Errorf("%s doesn't support negative ratio: %v. The ratio is the value of annotation key %s", launcherMutatorStr, ratio, annotationKey)
+	}
+
+	for i, container := range launcherPod.Spec.Containers {
+		request, requestExists := container.Resources.Requests[resourceName]
+		_, limitExists := container.Resources.Limits[resourceName]
+
+		if requestExists && !limitExists {
+			newQuantity := m.multiplyResource(request, ratio)
+			m.logInfo("Replacing old quantity (%s) with new quantity (%s) for pod %s/%s, UID: %s, accodring to ratio: %v",
```

Comment:
i think it would be nice to specify in the log if the quantity that was replaced is memory or cpu.